wgengine/magicsock/magicsock.go

```diff
@@ -1004,6 +1016,13 @@ func (c *Conn) determineEndpoints(ctx context.Context) (ipPorts []string, reason
 		}
 	}
 
+	if ext, err := c.portMapper.CreateOrGetMapping(ctx); err == nil {
```

Comment:
does magicsock guarantee that we determineEndpoints often enough to keep our mapping alive? This feels like a rake in the grass, where in a future change we'll optimize how we determineEndpoints and end up breaking our NAT mappings.

Reply:
Yeah, it should be fine. We determine endpoints regularly (every ~20-27s) when there's active traffic, and whenever a new connection starts up (for sending the CallMeMaybe frames, which need to know endpoints)